chemprop/cli/hpopt.py

```diff
@@ -227,8 +226,8 @@ def process_hpopt_args(args: Namespace) -> Namespace:
 
 
 def build_search_space(search_parameters: list[str], train_epochs: int) -> dict:
-    if "warmup_epochs" not in SEARCH_SPACE and "warmup_epochs" in search_parameters:
-        SEARCH_SPACE["warmup_epochs"] = tune.quniform(lower=1, upper=train_epochs // 2, q=1)
+    if "warmup_epochs" in search_parameters and SEARCH_SPACE.get("warmup_epochs", None) is None:
```

Comment:
```suggestion
    if "warmup_epochs" in search_parameters and SEARCH_SPACE.get("warmup_epochs", True):
```

Reply:
Ah good point! Not an equivalent suggestion

Reply:
Can you explain what happens when a custom range is provided  by the user? The `SEARCH_SPACE.get("warmup_epochs", True)` will evaluate to that range which will either give a syntax error while using with `if` or it may evaluate to `True` as it is a non-zero number which is not what we want. Please let me know! I'll try running some cases to better my understanding.

Reply:
It would result in True since a non-empty list is True, but don't worry about it, I merged without this suggestion based on your comment above.